src/main/java/com/cburch/logisim/file/XmlWriter.java

```diff
@@ -96,32 +98,20 @@ static String attrsToString(NamedNodeMap a) {
     return String.join(" ", lst);
   }
 
-  static int stringCompare(String a, String b) {
-    if (a == null) return -1;
-    if (b == null) return 1;
-    return a.compareTo(b);
+  private static final int stringCompare(String string1, String string2) {
+    if (string1 == null) return -1;
+    if (string2 == null) return 1;
+    return string1.compareTo(string2);
   }
   
-  static final void swap(Node top, Node nodeA, Node nodeB) {
-    final var nameNodeA = nodeA.getNodeName();
-    final var nameNodeB = nodeB.getNodeName();
-    var compare = stringCompare(nameNodeA, nameNodeB);
-    if (compare > 0) {
-      top.insertBefore(nodeB, nodeA);
-    } else if (compare < 0) return;
-    final var nameAttributesNodeA = attrsToString(nodeA.getAttributes());
-    final var nameAttributesNodeB = attrsToString(nodeB.getAttributes());
-    compare = stringCompare(nameAttributesNodeA, nameAttributesNodeB);
-    if (compare > 0) {
-      top.insertBefore(nodeB, nodeA);
-    } else if (compare < 0) return;
-    final var nameNodeValueNodeA = nodeA.getNodeValue();
-    final var nameNodeValueNodeB = nodeB.getNodeValue();
-    compare = stringCompare(nameNodeValueNodeA, nameNodeValueNodeB);
-    if (compare > 0) {
-      top.insertBefore(nodeB, nodeA);
-    }
-  }
+  private static final Comparator<Node> nodeComparator =
+      (a, b) -> {
```

Comment:
can we have `nodeA`, `nodeB` instead of `a`, `b`?

Reply:
Done